pkg/apis/workspaces/v1alpha2/component_compose.go

```diff
@@ -0,0 +1,38 @@
+package v1alpha2
+
+// ComposeFileComponentLocationType describes the type of
+// the location where the docker-compose file is fetched from.
+// Only one of the following types can be specified.
+// +kubebuilder:validation:Enum=Uri;Inlined
+type ComposeFileComponentLocationType string
+
+const ( 
+	UriComposeComponentLocationType ComposeFileComponentLocationType = "Uri"
+	InlinedComposeComponentLocationType ComposeFileComponentLocationType = "Inlined"
+)
+// +union
+type ComposeFileComponentLocation struct{
+	// Type of Compose File Component Location
+	// + 
+	// +unionDiscriminator
+	// +optional
+	LocationType ComposeFileComponentLocationType `json:"locationType,omitempty"`
+
+	// Location uri of the docker-compose file
+	// +optional
+	Uri string `json:"uri,omitempty"`
+
+	// Inlined Manifest of the docker-compose file
```

Comment:
Are docker-compose yaml files referred to as manifests? This reads as slightly confusing to me.
```suggestion
	// Inlined docker-compose file
```

Reply:
@amisevsk Yes, makes sense. Will make that change.